app/models/event.rb

```diff
@@ -20,18 +20,52 @@ class Event < ActiveRecord::Base
 
   accepts_nested_attributes_for :users
   accepts_nested_attributes_for :user_events
-  validates_presence_of :subject
+  validates_presence_of :subject, :datetime, :location
+  validate :datetime_no_less_than_15_mins_before_now
 
-  after_create :send_email
-  after_update :send_email
+  after_create :send_default_email
+  after_update :update_db, :send_updated_email
 
-  def send_email
+  def send_email(run_at_time, email_type)
     event = Event.find(id)
-    event_time = (event.datetime.to_time - 3.hours).to_datetime
-    users = User.all
-    users.each do |u|
-      UserMailer.delay(run_at: event_time)
-                .notification_email(u, event)
+    if event.try(:confirmed?)
+      users = User.all
```

Comment:
users variable is used only once. Can't we just remove it and use directly User.all.each ?